stack/source.go

```diff
@@ -270,8 +279,22 @@ func augmentCall(call *Call, f *ast.FuncDecl) {
 			call.Args.Processed = append(call.Args.Processed, fmt.Sprintf("%g", math.Float32frombits(uint32(pop()))))
 		case "float64":
 			call.Args.Processed = append(call.Args.Processed, fmt.Sprintf("%g", math.Float64frombits(pop())))
-		case "int", "int8", "int16", "int32", "int64", "uint", "uint8", "uint16", "uint32", "uint64":
+		case "uint", "uint8", "uint16", "uint32", "uint64":
 			call.Args.Processed = append(call.Args.Processed, fmt.Sprintf("%d", pop()))
+		// NOTE: we need a separate case per signed int type so that we can
+		// properly interpret negative integers.
+		case "int":
+			// Assume the stack was generated with the same bitness (32 vs 64)
+			// as the code processing it.
+			call.Args.Processed = append(call.Args.Processed, fmt.Sprintf("%d", int64(int(pop()))))
```

Comment:
The int64() cast is unnecessary.
I don't recall why I didn't use strconv.FormatInt() instead of fmt.Sprintf("%d") since it's slightly faster, but it's likely not a very hot path.